qcelemental: backport upstream patch

sheepforce opened the pull request:
###### Motivation for this change
Qcelemental constructs binary search paths wrong and therefore discards intended results silently. See https://github.com/MolSSI/QCElemental/pull/265

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

r-rmcgibbo commented:
Result of `nixpkgs-review pr 125396` at 2229d603 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.qcelemental</li>
    <li>python39Packages.qcelemental</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/development/python-modules/qcelemental/default.nix:1:39:
```
  |
1 | { buildPythonPackage, lib, fetchPypi, fetchpatch, numpy
  |                                       ^
```
</li>
  </ul>
</details>

markuskowa commented on pkgs/development/python-modules/qcelemental/default.nix:
```diff
@@ -14,10 +14,14 @@ buildPythonPackage rec {
 
+  # FIXME: Fixed upstream but not released yet. Nevertheless critical for correct behaviour.
+  # See https://github.com/MolSSI/QCElemental/pull/265
+  patches = [ ./SearchPath.patch ];
```
Please use `fetchpatch` here pull the diff directly from GitHub instead of adding an explicit patch file to nixpkgs (go to the PR look at the diff and add `.diff` to the URL to get a URL for the patch).

sheepforce replied:
I am afraid this can't be done here. There is an additional change to a file upstream to fix their CI. This file does not exist on the release version and therefore the patch from the PR fails.

markuskowa replied:
Doesn't commit 2211a4e59690bcb14265a60f199a5efe74fe44db and 5a32ce33e8142047b0a00d0036621fe2750e872a generate exactly your patch?
I.e. it is two URLs for fetchpatch then:
https://github.com/MolSSI/QCElemental/pull/265/commits/2211a4e59690bcb14265a60f199a5efe74fe44db.diff
https://github.com/MolSSI/QCElemental/pull/265/commits/5a32ce33e8142047b0a00d0036621fe2750e872a.diff

sheepforce replied:
Uh, i didn't know that this also works for individual commits but this makes a lot of sense. Has been changed.

sheepforce committed:
qcelemental: backport upstream patch

Qcelemental constructs binary search paths wrong and therefore discards intended results silently. See https://github.com/MolSSI/QCElemental/pull/265
qcelemental: use fetchpatch to get patches

markuskowa commented:
@GrahamcOfBorg build python3Packages.qcelemental
